scripts/spack_packages/umpire/package.py

```diff
@@ -118,290 +83,149 @@ class Umpire(CMakePackage, CudaPackage):
-                cfg.write(cmake_cache_option("ENABLE_DEVICE_CONST", True))
-
-            if using_toolchain:
-                cuda_flags.append("-Xcompiler {}".format(using_toolchain[0]))
-
-            cfg.write(cmake_cache_string("CMAKE_CUDA_FLAGS",  ' '.join(cuda_flags)))
+            if self.spec_uses_toolchain(self.spec):
+                cuda_flags.append("-Xcompiler {}".format(self.spec_uses_toolchain(self.spec)[0]))
 
-        else:
-            cfg.write(cmake_cache_option("ENABLE_CUDA", False))
+            entries.append(cmake_cache_string("CMAKE_CUDA_FLAGS",  ' '.join(cuda_flags)))
 
+        entries.append(cmake_cache_option("ENABLE_HIP", "+hip" in spec))
         if "+hip" in spec:
-            cfg.write("#------------------{0}\n".format("-" * 60))
-            cfg.write("# HIP\n")
-            cfg.write("#------------------{0}\n\n".format("-" * 60))
-
-            cfg.write(cmake_cache_option("ENABLE_HIP", True))
-
-#            -DHIP_ROOT_DIR=/opt/rocm-3.6.0/hip -DHIP_CLANG_PATH=/opt/rocm-3.6.0/llvm/bin
-
             hip_root = spec['hip'].prefix
             rocm_root = hip_root + "/.."
-            cfg.write(cmake_cache_entry("HIP_ROOT_DIR",
+            entries.append(cmake_cache_string("HIP_ROOT_DIR",
                                         hip_root))
-            cfg.write(cmake_cache_entry("HIP_CLANG_PATH",
+            entries.append(cmake_cache_string("HIP_CLANG_PATH",
                                         rocm_root + '/llvm/bin'))
-            cfg.write(cmake_cache_entry("HIP_HIPCC_FLAGS",
+            entries.append(cmake_cache_string("HIP_HIPCC_FLAGS",
                                         '--amdgpu-target=gfx906'))
-            cfg.write(cmake_cache_entry("HIP_RUNTIME_INCLUDE_DIRS",
+            entries.append(cmake_cache_string("HIP_RUNTIME_INCLUDE_DIRS",
                                         "{0}/include;{0}/../hsa/include".format(hip_root)))
             hip_link_flags = "-Wl,--disable-new-dtags -L{0}/lib -L{0}/../lib64 -L{0}/../lib -Wl,-rpath,{0}/lib:{0}/../lib:{0}/../lib64 -lamdhip64 -lhsakmt -lhsa-runtime64".format(hip_root)
             if '%gcc' in spec:
                 gcc_bin = os.path.dirname(self.compiler.cxx)
                 gcc_prefix = join_path(gcc_bin, '..')
-                cfg.write(cmake_cache_entry("HIP_CLANG_FLAGS",
-                "--gcc-toolchain={0}".format(gcc_prefix))) 
-                cfg.write(cmake_cache_entry("CMAKE_EXE_LINKER_FLAGS",
-                hip_link_flags + " -Wl,-rpath {}/lib64".format(gcc_prefix)))
+                entries.append(cmake_cache_string("HIP_CLANG_FLAGS", "--gcc-toolchain={0}".format(gcc_prefix))) 
+                entries.append(cmake_cache_string("CMAKE_EXE_LINKER_FLAGS", hip_link_flags + " -Wl,-rpath {}/lib64".format(gcc_prefix)))
             else:
-                cfg.write(cmake_cache_entry("CMAKE_EXE_LINKER_FLAGS", hip_link_flags))
+                entries.append(cmake_cache_string("CMAKE_EXE_LINKER_FLAGS", hip_link_flags))
```

Comment:
So, if I understand correctly, HIP is be a compiler not recognized as such is Spack (yet). So when we ask for `umpire%gcc +hip` under the hood we do `umpire%hip +gcc`. Correct?

Reply:
Yup, that's right. It's a bit like CUDA combo - and especially in Umpire since we mostly use the hip runtime library, we are still using a regular compiler to build most things.